internal/fwschemadata/data_set_at_path_test.go

```diff
@@ -2924,6 +2924,39 @@ func TestDataSetAtPath(t *testing.T) {
 				"other": tftypes.NewValue(tftypes.DynamicPseudoType, nil),
 			}),
 		},
+		"write-tftypes-value": {
```

Comment:
We should add a test for the error scenario of the `tftypes.Value` being the wrong type (like a number targeting a string)